### ISSUE TITLE
React 18 on setState the URL doesn't update

### ISSUE THREAD

**Issue by ariel-upstream** [xiel/use-location-state#375]
Hi, 
After I upgrade to react 18 it seems like the URL isn't updated if I try to update more than one state
For example: 

```
const [age, seAge] = useQueryState(age, 2)
const [names, seNames] = useQueryState(names, ["name1"])

const onSumbit = () => {
   setAge(5)
   setNames(["name1, name2"])
}

const onClear = () => {
   setAge(2)
   setName(["name1"])
}
```

After onSumbit the URL will change correctly to names=name1&names=name2 but after onClear it will be the same ( names=name1&names=name2 )

versions: 
"react-router-use-location-state": "^3.0.2",
"react": "18.1.0",
"react-router-dom": "6.1.0",

**Comment by xiel** [comment]
@ariel-upstream Can you add information what URL you currently see after the events?

**Comment by ariel-upstream** [comment]
@xiel I updated the example

**Comment by xiel** [comment]
@ariel-upstream You have u always pass an array (if the default value is an array).

setNames(["name1"])

**Comment by ariel-upstream** [comment]
Sure, it was ta typo
the same code works for me with "use-location-state": "~2.5.0"

**Comment by xiel** [comment]
@ariel-upstream Mh that's stange. I created a CodeSandbox, but it seems to work as expected with the new versions as well. Can you update the sandbox or create a sandbox which shows your problem?
https://codesandbox.io/s/hardcore-dream-yt6dex?file=/src/pages/QueryStateDemo.tsx

```
"react": "18.2.0",
"react-dom": "18.2.0",
"react-router-dom": "6.3.0",
"react-router-use-location-state": "3.0.2"
```

Cheers.

**Comment by peterlau** [comment]
I've run into this issue and can reproduce it in the codesandbox below: https://codesandbox.io/s/stupefied-estrela-i5x3pt?file=/src/pages/QueryStateDemo.tsx

Note that it seems to be specific to the case where an empty array is passed.

**Comment by oleksiy-letushev-axon** [comment]
I am facing the same weird issue. It looks like it happens when we deal with multiple query states and resetting to an empty value. Take a look at this example https://codesandbox.io/s/elegant-alex-en8bjz?file=/src/pages/QueryStateDemo.tsx. Setting a query state back to an empty array doesn't work. 

Though I noticed one interesting thing
```tsx
<button
  type="button"
  onClick={() => {
    setNames([]);
    setAges([]);
    setAppliedNames([]);
    // Removing this line fixes the problem. It doesn't clear why
    // setAppliedAges([]); 
  }}
>
  Reset filters
</button>
```

**Comment by xiel** [comment]
Thanks for the updated example. I'll have a look at it.

**Comment by xiel** [comment]
Thanks for opening the ticket @ariel-upstream and @oleksiy-letushev-axon for the updated demo.

The bug was fixed, please use the newly published version 📦 [v3.1.2](https://github.com/xiel/use-location-state/releases/tag/v3.1.2).

Here is the [demo](https://codesandbox.io/s/epic-snow-gpw82i?file=/src/pages/QueryStateDemo.tsx) with the fixed version.

Cheers.
